gst/nnstreamer/nnstreamer_plugin_api_filter.h

```diff
@@ -31,16 +31,44 @@ extern "C" {
 #endif
 
 /**
- * @brief nnapi hw properties.
+ * @brief acceleration hw properties.
+ * @details Enabling acceleration (choosing any accelerator hardware other
+ *          ACCL_NONE) will enable acceleration for the framework dependent on
+ *          the acceleration supported by the framework.
+ *
+ *          For example, enabling acceleration with tflite will enable NNAPI.
+ *          However, with NNFW will enable GPU/NEON etc.
+ *
+ *          Appropriate acceleration should be used with each framework. For
+ *          example, ACCL_NEON is only supported with NNFW tensor filter. Using
+ *          ACCL_NEON with pytorch would result in a warning message, and
+ *          the accelerator would fallback on ACCL_AUTO.
+ *
+ *          ACCL_AUTO automatically chooses the accelerator based on the ones
+ *          supported by the subplugin framework. However, ACCL_DEFAULT would
+ *          use the accelerator set by the subplugin framework, if any.
+ *
+ * @note Acceleration for a framework can be enabled/disabled in the build. If
+ *       the acceleration for a framework was disabled in the build, setting the
+ *       accelerator while use a tensor filter with that framework will have no
+ *       effect.
  */
 typedef enum
 {
-  NNAPI_CPU = 0,
-  NNAPI_GPU = 1,
-  NNAPI_NPU = 2,
+  /**< no explicit acceleration */
+  ACCL_NONE = 0,    /**< no acceleration (defaults to CPU) */
+
+  /** Enables acceleration */
+  ACCL_AUTO = 1,    /**< choose device automatically */
+  ACCL_CPU  = 2,    /**< specify device as CPU, if possible */
```

Comment:
Since there is the ACCL_NONE mode, ACCL_CPU might be misled that this means accelerations using SIMD. I suggest merging ACCL_CPU and  ACCL_NEON into ACCL_CPU_SIMD.

Reply:
As per [nnapi](https://developer.android.com/ndk/guides/neuralnetworks), NNAPI can fallback on CPU for some operations. However, that can be different from the frameworks default implementation for the CPU (https://github.com/tensorflow/tensorflow/issues/28283#issuecomment-488442568). So, ACCL_CPU denotes accelerators implementation for the CPU, and ACCL_NONE denotes direct support for the CPU by the framework itself.

Reply:
Note that currently, NNAPI does not allow choosing a particular device for acceleration as of now. ACCL_* is just to support this kind of mechanism, if needed in future.

Reply:
> As per [NNAPI](https://developer.android.com/ndk/guides/neuralnetworks), NNAPI can fallback on CPU for some operations. However, that can be different from the frameworks default implementation for the CPU ([tensorflow/tensorflow#28283 (comment)](https://github.com/tensorflow/tensorflow/issues/28283#issuecomment-488442568)). So, ACCL_CPU denotes accelerators implementation for the CPU, and ACCL_NONE denotes direct support for the CPU by the framework itself.

I am still confusing:) Are there any ways to accelerate NNs using CPU without SIMD? 
- ACCL_NONE: just CPU
- ACCL_CPU: acceleration using whatever in CPU
- ACCL_NEON: acceleration using ARM SIMD in ARM CPU 
?

Give me more intuition, please.

Reply:
1. What if a user wants to express "Use either GPU or NPU. Try to avoid using CPU" or "Use NPU-2 for this filter, NPU-1 for that filter."?

2. Can we make this aligned with /api/capi/include/nnstreamer.h::ml_nnfw_hw_e ?

Reply:
> I am still confusing:) Are there any ways to accelerate NNs using CPU without SIMD?
> 
> * ACCL_NONE: just CPU
> * ACCL_CPU: acceleration using whatever in CPU
> * ACCL_NEON: acceleration using ARM SIMD in ARM CPU
>   ?
> 
> Give me more intuition, please.

My bad, I should have made this more clearer - 
- ACCL_NEON: specific for nnfw backend for now. Will represent SIMD if any framework supports it.
- ACCL_NONE: just CPU
- ACCL_CPU: this is still CPU, but now the backend of CPU will use the implementation provided with the acceleration library supported by the framework. For eg - if a framework provides acceleration for CPU using a custom-made BLAS lib, then it would fall in this category. 

Whether they use SIMD internally or not, cant say. However, if some acceleration states that they have explicitly used SIMD (like nnfw `acl_neon` vs `cpu` mode), then ACCL_SIMD could be used.

Another example where ACCL_CPU and ACCL_NONE would be different is with tflite running in a system with no GPU/NPU/DSP support for NNAPI - 
- If NNAPI is used, then tflite would use the NNAPI CPU implementation - ACCL_CPU
- if NNAPI is not used, then tflite would use its own CPU implementation - ACCL_NONE
In this case, if NNAPI gives the option to choose the backend to run (sometime in future), then IMO ACCL_CPU would differ from ACCL_NONE.

Please correct me if I am wrong.

Reply:
> 1. What if a user wants to express "Use either GPU or NPU. Try to avoid using CPU" or "Use NPU-2 for this filter, NPU-1 for that filter."?

I think for this, input can be taken as a regular expression and then interpreted correspondingly.
Like `true:(NPU,GPU,!(CPU))`. 

> "Use NPU-2 for this filter, NPU-1 for that filter."

In this case, as the accelerator property is set for each filter in the pipeline independently, I think this scenario is automatically covered. 

> 2. Can we make this aligned with /api/capi/include/nnstreamer.h::ml_nnfw_hw_e ?

Updated the values for the `enum accl_hw`. Most of the attributes align. However, can `ML_NNFW_HW_ANY` be covered with `ACCL_DEFAULT` or should `ACCL_ANY` be introduced?